jdaviz/core/template_mixin.py

```diff
@@ -1677,6 +1679,69 @@ def _check_dataset_spectral_subset_valid(self, event={}, return_ranges=False):
             return self.spectral_subset_valid
 
 
+class NonFiniteUncertaintyMismatchMixin(VuetifyTemplate, HubListener):
+    """ Adds a traitlet that identifies if there are any finite data values
+        that correspond to a non-finite uncertainty at that index.
+
+        In model fitting, the presence of finite, fittable data with corresponding
+        non-finite uncertainties can cause issues. Finite data values will be
+        filtered out in this case which may be undesirable. This traitlet when
+        True triggers a warning in the model fitting plugin (in specviz only,
+        currently) if there are any finite values with non-finite uncertainties.
+
+        Note that if a the uncertainty array is FULLY non-finite and the data is
+        FULLY finite, uncertainties will be set to None (in the Specviz parser),
+        so this traitlet will be False in that case (and therefore no warning
+        message displayed in the plugin).
+    """
+
+    non_finite_uncertainty_mismatch = Bool(False).tag(sync=True)
+
+    # every time a data/subset selection is changed, check the data selection and
+    # its uncertainties to see if there are any finite data elements with
+    # uncertainties. Warn in plugin if this occurs.
+
+    # note! the 'n/a' traitlet doesn't exist. we are watching it here because
+    # DatasetSpectralSubsetValidMixin is also watching the exact set
+    # of real traitlets, which creates a bug where only the last inhereted method
+    # will run, so we need to spoof it like this.
+    @observe("n/a", "dataset_selected", "spectral_subset_selected")
+    def _check_non_finite_uncertainty_mismatch(self, event={}):
+
+        if not hasattr(self, 'dataset') or self.dataset_selected == '':
+            # during initial init, this can trigger before the component is initialized
+            return
+        try:
+            spec = self._get_1d_spectrum()  # this is a method in model fitting
+        except AttributeError:
+            # only model_fitting has _get_1d_spectrum(), but this method is here
+            # instead of there  because it may eventually be used by other plugins.
+            # if that happens, move _get_1d_spectrum() somewhere more general
+            raise ValueError('_get_1d_spectrum() is only implemented in model fitting.')
+
+        flux = spec.flux
+        uncert = spec.uncertainty
```

Comment:
Do we want to account for the selected _spectral_ subset (so that if the selected subset manages to avoid the nans, the user doesn't see the warning)?

Reply:
how/where?  As far as I can tell, `_get_1d_spectrum()` accounts for the spatial, but not spectral, subset selection.

Reply:
i did what i _thought_ would select the spectral subset (if Entire Spectrum, then you should be able to use _get_spectrum_1d() even if its specviz. if not, then use self.dataset.selected_obj. however, this isn't working. when i select a spectral subset (either in the viewer, or in the plugin, i wasn't sure which 'selected_obj' this was referring to i was just following the same logic used in the DatasetSpectralSubsetValidMixin class above), .dataset.selected_obj always points to the entire spectrum.

Reply:
ok this should work now!